File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
s = input()
numbers = []
strings = []
add_number = ""

for number in s:
    if number.isnumeric():
        add_number += number
    else:
        numbers.append(int(add_number))
        add_number = ""
        strings.append(number)
numbers.append(int(add_number))


for i in range(1, len(numbers)):
    if strings[i - 1] == "-":
        numbers[i] = numbers[i] * -1

answer = 0
flag = False
top_value = 0

for i in range(len(numbers)):
    if not flag:
        if numbers[i] > 0:
            answer += numbers[i]
        else:
            flag = True
            answer += numbers[i]
    else:
        if numbers[i] > 0:
            answer -= numbers[i]
        else:
            answer += numbers[i]
print(answer)    

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.js
let fs = require("fs");

let input = fs.readFileSync("/dev/stdin").toString().split("\n");


let strings = input[0];

let newStrings = strings.split("-");
let values = [];
for(let i = 0; i < newStrings.length; i++){
  let news = newStrings[i].split("+").map(Number);
  let sum = news.reduce((a, b) => a + b);
  values.push(sum);
}

if(values.length === 1){
  console.log(values[0]);
}
else{
  let result = values.reduce((a, b) => a + b);
  console.log(values[0] * 2 - result);
}